src/metrics/metrics_manager.py

```diff
@@ -23,25 +20,37 @@ def __init__(self, metric: Metric, mapping: Dict[str, str], log_name: str):
             log_name: The metric name used in logs.
         """
         super().__init__()
-        self.__metric = metric
-        self.__mapping = mapping
-        self.__log_name = log_name
+        self._metric = metric
+        self._mapping = mapping
+        self._log_name = log_name
+
+    @property
+    def metric(self) -> Metric:
+        """The metric."""
+        return self._metric
 
     @property
     def log_name(self) -> str:
         """The metric name used in logs."""
-        return self.__log_name
+        return self._log_name
 
     @property
     def mapping(self) -> Dict[str, str]:
         """Dictionary for mapping Metric forward input keys with Task output dictionary keys."""
-        return self.__mapping
-
+        return self._mapping
+    
     def forward(self, *args, **kwargs):
-        return self.__metric(*args, **kwargs)
+        """Update metric states."""
```

Comment:
Is documentation still needed? Also, do we still need forward if it calls `update`?

Reply:
Yes, we need forward because it's nn.Module